src/win32/posix_w32.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_w32_posix_h__
```

Comment:
I was going to complain that this doesn't accurately match the filename, but I think that the filename is wrong here.  This doesn't seem to have much to do with POSIX compliance, and if it did, I would rather see it in `posix.h` instead of a similarly named file.

I would very much expect to find `git_win32_path_*` functions inside `path_w32.c`.  I would very much expect the decls for those functions in `path_w32.h`.  Can we move that and drop this new file?